src/app/components/ItemFilters/ItemFilters.jsx

```diff
@@ -45,21 +45,34 @@ const ItemFilters = (
   }, [numOfFilteredItems, parsedFilterSelections]);
 
   /**
-   * When new filters are selected or unselected, fetch new items.
+   * When filters are appied or cleared, build new filter query
    */
-  const buildFilterUrl = (clear = false, clearYear = false) => {
-    let queryObj = {};
-    if (!clear) {
-      Object.keys(selectedFields).filter(field => selectedFields[field].length).forEach(field => {
-        const selectedFilterValues = selectedFields[field].join(',')
+  const buildFilterQuery = (clear = false, clearYear = false) => {
+    let queryObj = {}
+    let fieldsToQuery
+    // clear is only true when we are clearing all filters, so return 
+    //   empty query object
+    if (clear === true) return queryObj
+    // clear equals false indicates a filter is being applied
+    if (clear === false) {
+      fieldsToQuery = selectedFields
+      // clear is a field, to indicate which field we want to clear
+    } else if (clear.length) {
```

Comment:
Okay, I see why the above is `clear === true`. Make a new function param for this so we can have `clear` just be a boolean and have another param, probably called `field`, for this other functionality. It'll be clearer than one function that can be true, false, or an array.

Reply:
updated